api/lib/validation/attributeValidator.js

```diff
@@ -20,6 +20,19 @@ var compareDateTimes = function(startDateTime,endDateTime) {
   return moment(startDateTime,'YYYY-MM-DDTHH:mm')
     .isSameOrAfter(moment(endDateTime,'YYYY-MM-DDTHH:mm'));
 }
+var normalizeStartDate = function(object) {
```

Comment:
why we do normalization, start_date and end_date are required fields, aren't they?

Reply:
@boyang9527 , start_date and end_date are optional fields, so it is allowed to set "start_date" or "end_date" only, the function "normalizeStartDate" and "normalizeEndDate" facilitates  the time overlapping detection if "start_date" or "end_date" not set.